core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -220,6 +222,14 @@ public BigQuerySqlDialect(SqlDialect.Context context) {
       writer.literal("INT64");
       writer.endFunCall(castFrame);
       break;
+    case SYSTEM_FUNCTION:
+      if (call.getOperator().equals(CURRENT_USER)) {
```

Comment:
This seems to be a common code in all 3 dialects. Move this unparse logic to sqlLibraryOperators